source/grammar/index.rst

```diff
@@ -80,53 +88,59 @@ repetition is denoted by curly brackets ``+`` (one or more) and ``*`` (zero or m
     incrementor: "++" | "--
     pragma: "#pragma {" ... "}"
     declaration_statement: (quantum_declaration | classical_declaration | constant_declaration) ";"
-    assignment_expression: expression assignment_operator expression
-    binary_operator: "+" | "-" | "*" | "/" 
+    assignment_expression: "=" expression
+    binary_operator: "+" | "-" | "*" | "/"
         :| "<<" | ">>" | "rotr" | "rotl" | "&" | "|" | "^" | "&&" | "||"
         :| ">" | "<" | ">=" | "<=" | "==" | "!="
     assignment_operator: "=" | "+=" | "-=" | "*=" | "/=" |
         :| "<<=" | ">>=" |  "&=" | "|=" | "^=" | "~=" | "->"
     membership_test: id "in" set_declaration
-    index_set_declaration: "{" index_set_generators "}"
-    index_set_generators: expression_list | range
-    range: expression? ":" expression? ( ":" expression )?
+    index_set_declaration: "{" expression_list "}" | range
+    range: "[" expression? ":" expression? ( ":" expression )? "]"
     selection_statement: "if (" expression ")" program_block ( "else" program_block )?
-    iteration_statement: "for" membership_test program_block | "while (" expression ")" program_block
+    iteration_statement: ( "for" membership_test | "while (" expression ")" ) program_block
     selection_directive_statement: selection_directive ";"
     selection_directive: "break" | "continue" | "exit"
     quantum_gate_definition: "gate" quantum_gate_signature quantum_gate_block
-    quantum_gate_signature: id quantum_gate_args id_lis
-    quantum_gate_args: ( "(" classical_type_and_id_list? ")" )?
+    quantum_gate_signature: id classical_params id_list
+    id_list: ( id "," )* id
+    index_type: id ( "[" expression "]" )?
+    index_list: ( index_type "," ) index_type
+    classical_params: ( "(" classical_type_and_id_list? ")" )?
     quantum_gate_block: "{" quantum_gate_call* "}"
     quantum_type_and_id_list: ( quantum_type_and_id "," )* quantum_type_and_id
-    quantum_type_and_id: quantum_type designator? association 
+    quantum_type_and_id: quantum_type designator? association
     quantum_statement: quantum_instruction ";"
     quantum_instruction: quantum_gate_call
         :| quantum_measurement
-    quantum_measurement: "measure" argument
+        :| barrier_instruction
+    quantum_measurement: "measure" quantum_declaration "->" bit_declaration
+        :| bit_declaration "= measure" quantum_declaration
```

Comment:
```suggestion
    quantum_measurement: "measure" quantum_declaration "->" bit_declaration
        :| bit_declaration "="  "measure" quantum_declaration
```
Quantum measurement assignment should be built into an expression as I should be able to `int[32] a = int[32](measure qubit)`. IT is fine as it is above already and a separate declaration for the deprecated `->` notation should be added `"measure" quantum_declaration "->" argument`